Tag repo with Go compatible version names

jasonwzm opened the pull request:
fix https://github.com/istio/api/issues/1201

The fix does not do double tagging. If a repo needs to be exposed for external use, we will just release it with go version convention.

istio-testing commented:
@jasonwzm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
dry-run_release-builder | 93396be7d4d266fa2de8a3443b7c2af51a505e79 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/dry-run_release-builder/24) | `/test dry-run_release-builder`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jasonwzm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
dry-run_release-builder | 93396be7d4d266fa2de8a3443b7c2af51a505e79 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/dry-run_release-builder/24) | `/test dry-run_release-builder`
gencheck_release-builder | 93396be7d4d266fa2de8a3443b7c2af51a505e79 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/gencheck_release-builder/55) | `/test gencheck_release-builder`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jasonwzm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
dry-run_release-builder | 93396be7d4d266fa2de8a3443b7c2af51a505e79 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/dry-run_release-builder/24) | `/test dry-run_release-builder`
gencheck_release-builder | c7a4d809a29d2d5682bca70a48ab67141b5f140b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/gencheck_release-builder/56) | `/test gencheck_release-builder`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jasonwzm: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
gencheck_release-builder | c7a4d809a29d2d5682bca70a48ab67141b5f140b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/gencheck_release-builder/56) | `/test gencheck_release-builder`
dry-run_release-builder | c7a4d809a29d2d5682bca70a48ab67141b5f140b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/dry-run_release-builder/25) | `/test dry-run_release-builder`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@jasonwzm: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
dry-run_release-builder | d8740d1de364769be9c0f019915e6b893c33472d | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_release-builder/100/dry-run_release-builder/26) | `/test dry-run_release-builder`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented on pkg/manifest.go:
```diff
@@ -107,5 +108,8 @@ func ReadInManifest(manifestFile string) (model.InputManifest, error) {
 	if err := validateManifestDependencies(manifest.Dependencies); err != nil {
 		return manifest, fmt.Errorf("invalid manifest: %v", err)
 	}
+	if _, err = semver.StrictNewVersion(manifest.Version); err != nil {
```
I think this will fail for 1.5-dev.sha

jasonwzm replied:
can we name the release 1.5.0-dev.sha? So that all of our versions in the future will be semver.

howardjohn commented:
it's not accurate then. Consider the 1.4 cases. If we named a PR at head of
1.4 to be 1.4.0 then it's wrong, it's after 1.4.2. But we can't just name
it 1.4.3 since we don't know patch release info.

Simply adding patch release info to the repo won't work either since commit
order is not necessarily release order (confusingly) when we have security
patches in play, which is often. I can point you to a slack thread with a
lot of info if you are interested.

If there is a good solution though that would be great

On Mon, Dec 16, 2019, 10:00 AM Jason Wang <notifications@github.com> wrote:

> *@jasonwzm* commented on this pull request.
> ------------------------------
>
> In pkg/manifest.go
> <https://github.com/istio/release-builder/pull/100#discussion_r358380810>:
>
> > @@ -107,5 +108,8 @@ func ReadInManifest(manifestFile string) (model.InputManifest, error) {
>  	if err := validateManifestDependencies(manifest.Dependencies); err != nil {
>  		return manifest, fmt.Errorf("invalid manifest: %v", err)
>  	}
> +	if _, err = semver.StrictNewVersion(manifest.Version); err != nil {
>
> can we name the release 1.5.0-dev.sha? So that all of our versions in the
> future will be semver.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/release-builder/pull/100?email_source=notifications&email_token=AAEYGXITFN74FTJLPQJRFT3QY6645A5CNFSM4J3FFYWKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPKXVTY#discussion_r358380810>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXLNCVT4CEAUWIREABLQY6645ANCNFSM4J3FFYWA>
> .
>

jasonwzm commented:
@howardjohn If you can point me to those conversations, that'd be great! But from my understanding, with our current branching strategy, any new PRs that are merged will belong to current patch version + 1. For example, any new PR merged now will be built with version 1.4.3-dev.sha?

jasonwzm commented:
@howardjohn Thanks! Based on the scenarios discussed in the slack chat, I removed the semver check for the input version and moved it to where we tag go module version.
